### PR TITLE
Listener (WS) refactoring

### LINKED ISSUE #569: Aggregate bonded partialAdd ws listener filter issue
#### Issue: 
- Listener channels currently only filter on 'signer' and 'recipientAddress' in TransferTransaction only.
- Does not resolve alias (unresolvedAddress). e.g. Transfer's recipientAddress is an instance of NamespaceId

### PR DESCRIPTION
**Pull request by rg911**
#### Issue: 
- Listener channels currently only filter on 'signer' and 'recipientAddress' in TransferTransaction only.
- Does not resolve alias (unresolvedAddress). e.g. Transfer's recipientAddress is an instance of NamespaceId

#### Fixed:
- Added abstract method in Transactions for listener notification filtering purposes. 
- Filters on address / publicAccount / publicKeys on all transactions. e.g. 'targetAccount' in MetadataTransactions.
- Added `NamespaceRepository` to `Listener` to resolve alias.
Fixed #569

### CODE REVIEW

File: src/model/transaction/TransferTransaction.ts [hunk 1]
```diff
@@ -315,4 +315,19 @@ export class TransferTransaction extends Transaction {
...
     }
+
+    /**
+     * @internal
+     * Check a given address should be notified in websocket channels
+     * @param address address to be notified
+     * @param alias address alias (names)
+     * @returns {boolean}
+     */
+    public NotifyAccount(address: Address, alias: NamespaceId[]): boolean {
```
Comment (dgarcia360) [inline src/model/transaction/TransferTransaction.ts:326]: applies to all ``NotifyAccount`` functions:

- If it's tagged as @internal, shouldn't it be a private method?
- I would expect NotifyAccount to be notifyAccount (camelCase)

File: src/infrastructure/Listener.ts [hunk 2]
```diff
@@ -336,18 +316,76 @@ export class Listener implements IListener {
...
+            );
+        };
+    }
+
+    /**
+     * Checks if the transaction is signer by an address.
+     * @param transition the transaction
+     * @param address the address.
+     */
+    private isSigned(transition: Transaction, address: Address): boolean {
```
Comment (dgarcia360) [inline src/infrastructure/Listener.ts:385]: Does it make sense to move this function within the Transaction model?

Comment (dgarcia360) [inline src/infrastructure/Listener.ts:385]: typo in transition --> transaction

**Comment by dgarcia360** [comment]
@rg911 I like the approach to check also for the aliases assigned.to the account. Good idea adding a filter by transaction hash - looks handy to check if a given transaction has been confirmed or not.

File: src/model/transaction/TransferTransaction.ts (hunk 1 above)
Comment (fboucquez) [reply src/model/transaction/TransferTransaction.ts:326]: it should be lowe case

File: src/infrastructure/Listener.ts (hunk 2 above)
Comment (fboucquez) [reply src/infrastructure/Listener.ts:385]: Yes, transaction NotifyAccount should also call this method

File: src/model/transaction/TransferTransaction.ts (hunk 1 above)
Comment (rg911) [reply src/model/transaction/TransferTransaction.ts:326]: Internal should be fine, Internal won't work on abstract. will update the method name 👍

File: src/model/transaction/AccountAddressRestrictionTransaction.ts [hunk 3]
```diff
@@ -236,9 +236,9 @@ export class AccountAddressRestrictionTransaction extends Transaction {
      * @param alias address alias (names)
      * @returns {boolean}
      */
-    public NotifyAccount(address: Address, alias: NamespaceId[]): boolean {
```
Comment (dgarcia360) [inline src/model/transaction/AccountAddressRestrictionTransaction.ts]: (typo) shouldNotifiAccount --> shouldNotifYAccount

**Pull request merged**